cloud/google/README.md

```diff
@@ -0,0 +1,66 @@
+# Cluster API GCP Cloud Provider
+
+## Cluster Deletion
+
+This guide explains how to delete all resources that were created as part of
+your Cluster API Kubernetes cluster.
+
+If your cluster was created using the gcp-deployer tool, see the 
+[gcp-deployer docs](../../gcp-deployer/README.md).
+
+1. Remember the service accounts that were created for your cluster
+
+   ```bash
+   export MASTER_SERVICE_ACCOUNT=$(kubectl get cluster -o=jsonpath='{.items[0].metadata.annotations.gce\.clusterapi\.k8s\.io\/service-account-k8s-master}')
+   export WORKER_SERVICE_ACCOUNT=$(kubectl get cluster -o=jsonpath='{.items[0].metadata.annotations.gce\.clusterapi\.k8s\.io\/service-account-k8s-worker}')
+   export INGRESS_CONTROLLER_SERVICE_ACCOUNT=$(kubectl get cluster -o=jsonpath='{.items[0].metadata.annotations.gce\.clusterapi\.k8s\.io\/service-account-k8s-ingress-controller}')
+   export MACHINE_CONTROLLER_SERVICE_ACCOUNT=$(kubectl get cluster -o=jsonpath='{.items[0].metadata.annotations.gce\.clusterapi\.k8s\.io\/service-account-k8s-machine-controller}')
+   ```
+
+1. Remember the name and zone of the master VM
+
+   ```bash
+   export MASTER_VM_NAME=$(kubectl get machines -l set=master | awk '{print $1}' | tail -n +2)
+   export MASTER_VM_ZONE=$(kubectl get machines -l set=master -o=jsonpath='{.items[0].metadata.annotations.gcp-zone}')
+   ```
+
+1. Delete all of the node Machines in the cluster. Make sure to wait for the
+corresponding Nodes to be deleted before moving onto the next step. After this
+step, the master node will be the only remaining node.
+
+   ```bash
+   kubectl delete machines -l set=node
+   kubectl get nodes
+   ```
+
+1. Delete any Kubernetes objects that may have created GCE resources on your
+behalf, make sure to run these commands for each namespace that you created:
+
+   ```bash
+   --- Ingress objects create GCE load balancers, including 
+   kubectl delete ingress --all
+   kubectl delete svc --all
+   kubectl delete pvc --all
+   ```
+
+   - Services can create a GCE load balancer if the type of the service is
```

Comment:
i'm confused by these three bullet points -- are they telling me to take action or are they justifying the three delete commands above related to load balancers? If they are just for justification I'd recommend just removing?

Reply:
I added these to address some earlier feedback about it not being clear which resources would be deleted. I moved them into comments above in the bash section, does that make it more clear?

Reply:
Yes, much more clear. I would say we don't need to document quite this much, but now that you've done so it is definitely nice! Thank you.